src/replication/ng/scheduler/scheduler.go

```diff
@@ -0,0 +1,27 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package scheduler
+
+import (
+	"github.com/goharbor/harbor/src/replication/ng/model"
+)
+
+// Scheduler schedules tasks to transfer resource data
+type Scheduler interface {
+	// Schedule tasks for one execution
+	Schedule([]*model.Resource, []*model.Resource) ([]*model.Task, error)
+	// Stop the task specified by ID
+	StopTask(id string) error
```

Comment:
Is it conflicted with PR https://github.com/goharbor/harbor/pull/6886? Can we use just `Stop` as the method name?